packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -220,7 +221,7 @@ const getFilterCallbackFromItem = (
         if (ignoreDiacritics) {
           value = removeDiacritics(value);
         }
-        return applyFilterOnRow(value, row, column, apiRef);
+        return applyFilterOnRow(value, row, column, getPublicApiRef(apiRef));
```

Comment:
We were passing private apiRef here

Reply:
Oops, nice catch 👍

Reply:
Can we create the API ref outside this function? I think this one is inside the filter loop, so we're doing `O(n)` allocations if we call `getPublicApiRef` here.

Reply:
Same comment applies below, line 420.

Reply:
Thanks for catching it, updated it!